lib/ecto/query/inspect.ex

```diff
@@ -27,7 +33,13 @@ defimpl Inspect, for: Ecto.Query do
         string
     end)
 
-    surround_many("#Ecto.Query<", list, ">", opts, fn str, _ -> str end)
+    # TODO: Replace unquote with just `container_doc` when Elixir < 1.8.0 is no longer supported
+    unquote(
+      if(Version.match?(System.version(), "~> 1.8.0-dev"),
+        do: quote(do: :container_doc),
+        else: quote(do: :surround_many)
+      )
+    )("#Ecto.Query<", list, ">", opts, fn str, _ -> str end)
```

Comment:
Can you please move the logic to outside of the function? Something like this:

```
    container_doc =  if Version.match?(System.version(), ">= 1.6.0"), do: :container_doc, else: :surround_many
```

Note the proper version to check is >= 1.6.0 and you don't need to quote atoms. Thanks!

Reply:
Sure, also as a note, doing `">= 1.6.0"` caused it to fail on the RC.1 and so forth, are those essentially pretended to not exist then?  If so that's fine, just want to be explicit.  :-)

Reply:
> Sure, also as a note, doing ">= 1.8.0" caused it to fail on the RC.1 and so forth

RC.1 of what? This function exists since v1.6.0 :)

Reply:
Pushed those changes with the `">= 1.8.0"` ignoring test 1.8.0 versions.  :-)

Reply:
Sorry, I think something is getting lost in translation. Let me try again: why not use `>= 1.6.0` (one-six-oh) instead of `>= 1.8.0` (one-eight-oh)?

Reply:
Oh was it introduced in 1.6.0?!  Ah so it was!  I was running a *really* old version of elixir on the CI until recently...  ^.^;

Updating!  Few minutes...  :-)